src/astx/literals/collections.py

```diff
@@ -0,0 +1,161 @@
+"""ASTX literal representations of collection types."""
+
+from __future__ import annotations
+
+from typing import Dict, List, Set, Tuple, Type
+
+from public import public
+
+from astx.base import AST, NO_SOURCE_LOCATION, SourceLocation
+from astx.literals.base import Literal
+from astx.tools.typing import typechecked
+from astx.types.base import AnyType
+from astx.types.collections import ListType, MapType, SetType, TupleType
+
+
+@public
+@typechecked
+class LiteralList(Literal):
+    """Literal representation of a list."""
+
+    elements: List[Literal]
+
+    def __init__(
+        self,
+        elements: List[Literal],
+        loc: SourceLocation = NO_SOURCE_LOCATION,
+    ) -> None:
+        super().__init__(loc)
+        self.elements = elements
+
+        # Collect the unique type classes from the elements
+        type_classes: Set[Type[AST]] = {type(elem.type_) for elem in elements}
```

Comment:
AST is very generic .. it is used for expressions and statements. and for all kind of data you cannot use a statement as a value. for example, you cannot do something like `mylist = [def my_func(): pass]`, so please don't use AST for anything related to data. You can use Expr, DataType, DataTypeOp, or ExprType, according to the necessity.